src/guides/v2.3/extension-dev-guide/admin-grid.md

```diff
@@ -0,0 +1,601 @@
+         </item>
+       </argument>
+   </argument>
+   <argument name="data" xsi:type="array">
+     <item name="js_config" xsi:type="array">
+        <item name="component" xsi:type="string">Magento_Ui/js/grid/provider</item>
+     </item>
+   </argument>
+  </dataSource>
+  <listingToolbar name="listing_top">
+    <bookmarks name="bookmarks" class="Dev\Grid\Ui\Component\Category\Listing\Bookmark"/>
+    <columnsControls name="columns_controls"/>
+    <massaction name="listing_massaction">
+      <argument name="data" xsi:type="array">
+        <item name="data" xsi:type="array">
+           <item name="selectProvider" xsi:type="string">dev_grid_category_listing.dev_grid_category_listing.dev_grid_category_columns.ids</item>
+           <item name="displayArea" xsi:type="string">bottom</item>
+           <item name="component" xsi:type="string">Magento_Ui/js/grid/tree-massactions</item>
+           <item name="indexField" xsi:type="string">entity_id</item>
+        </item>
+      </argument>
+      <action name="delete">
+         <argument name="data" xsi:type="array">
+           <item name="config" xsi:type="array">
+               <item name="type" xsi:type="string">delete</item>
+               <item name="label" xsi:type="string" translate="true">Delete</item>
+               <item name="url" xsi:type="url" path="dev_grid/category/massDelete"/>
+               <item name="confirm" xsi:type="array">
+                  <item name="title" xsi:type="string" translate="true">Delete items</item>
+                  <item name="message" xsi:type="string" translate="true">Are you sure you want to delete selected items?</item>
+               </item>
+           </item>
+         </argument>
+      </action>
+    </massaction>
+    <filters name="listing_filters">
+            <argument name="data" xsi:type="array">
+                <item name="config" xsi:type="array">
+                    <item name="templates" xsi:type="array">
+                        <item name="filters" xsi:type="array">
+                            <item name="select" xsi:type="array">
+                                <item name="component" xsi:type="string">Magento_Ui/js/form/element/ui-select</item>
+                                <item name="template" xsi:type="string">ui/grid/filters/elements/ui-select</item>
+                            </item>
+                        </item>
+                    </item>
+                </item>
+            </argument>
+    </filters>
+    <paging name="listing_paging"/>
+  </listingToolbar>
+  <columns name="dev_grid_category_columns">
+    <selectionsColumn name="ids">
+       <argument name="data" xsi:type="array">
+           <item name="config" xsi:type="array">
+              <item name="indexField" xsi:type="string">entity_id</item>
+           </item>
+       </argument>
+    </selectionsColumn>
+    <column name="entity_id">
+      <argument name="data" xsi:type="array">
```

Comment:
It's not critical, but I would recommend using the same approach for defining the column's settings for better consistency.

What I mean, is that technically it's possible to provide settings either by `<settings>` node or by `<argument ‌name="data"` element. Both of them will work. Nonetheless, the latter should be used only in case you need to apply a custom settings tree.

Within the sample code snippet, both approaches are used simultaneously. So I would recommend utilizing the same approach for all columns to avoid misunderstanding.
E.g.
```
    <column name="entity_id">
            <settings>
                <filter>textRange</filter>
                <label translate="true">ID</label>
                <resizeDefaultWidth>25</resizeDefaultWidth>
            </settings>
        </column>
```

Or please add some explanation below the example, so it will be clear why the different approaches are used.

Thank you!

Reply:
done.